plugin/pkg/admission/jsonpath/admission.go

```diff
@@ -0,0 +1,211 @@
+	} else {
+		fieldSpec = submatches[2]
+	}
+	return fmt.Sprintf("{.%s}", fieldSpec), nil
+}
+
+type JSONPathAdmissionRule struct {
+	// Version indicates the version for the rule.  For now this has to be "v1"
+	Version string
+
+	// Name is the name of this rule
+	Name string
+
+	// KindRegexp holds a regular expression used to determine which Kinds of
+	// objects this rule applies to.
+	KindRegexp string `json:"kindRegexp", yaml:"kindRegexp"`
+
+	// Path holds a JSONPath expression to select particular fields
+	Path string `json:"path", yaml:"path"`
+
+	// MatchRegexp holds a matching expression, all values returned by the JSONPath expression (above)
+	// must match this expression for the object to be admitted
+	MatchRegexp string `json:"matchRegexp", yaml:"matchRegexp"`
+
+	// AcceptEmptyMatch indicates if an empty match is acceptable.  Empty match means that the
+	// JSONPath expression above returns no values.  Default is false, meaning that at least one
+	// value that matches MatchRegexp is required for admission
+	AcceptEmptyMatch bool `json:"acceptEmptyMatch", yaml:"acceptEmptyMatch"`
+}
+
+type JSONPathAdmissionConfig struct {
+	Rules []JSONPathAdmissionRule `json:"rules", yaml:"rules"`
+}
+
+func init() {
+	admission.RegisterPlugin("jsonpath", func(client clientset.Interface, config io.Reader) (admission.Interface, error) {
+		if config == nil {
+			return nil, errors.New("Config file is required for JSONPath Admission Controller")
+		}
+		jsonConfig := JSONPathAdmissionConfig{}
+		data, err := ioutil.ReadAll(config)
+		if err != nil {
+			return nil, err
+		}
+		jsonData, err := yaml.ToJSON(data)
+		if err != nil {
+			return nil, err
+		}
+		if err := json.Unmarshal(jsonData, &jsonConfig); err != nil {
+			return nil, err
+		}
+		return NewJSONPathAdmission(jsonConfig)
+	})
+}
+
+type jsonPathRule struct {
+	name            string
+	kindRE          *regexp.Regexp
+	path            *util_jsonpath.JSONPath
+	matchRE         *regexp.Regexp
+	acceptNoMatches bool
+}
+
+func (r *jsonPathRule) String() string {
+	return fmt.Sprintf("%s: %s %s %s %v", r.name, r.kindRE.String(), r.path, r.matchRE.String(), r.acceptNoMatches)
+}
+
+func (r *jsonPathRule) admit(a admission.Attributes) error {
+	if !r.kindRE.MatchString(a.GetKind().Kind) {
+		return nil
+	}
+
+	vals, err := r.path.FindResults(a.GetObject())
+	if err != nil {
+		return err
+	}
+	resultCount := 0
+	buff := &bytes.Buffer{}
+	r.path.Execute(buff, a.GetObject())
+	for _, arr := range vals {
+		resultCount += len(arr)
+		for _, val := range arr {
+			text, err := r.path.EvalToText(val)
+			if err != nil {
+				return err
+			}
+			if !r.matchRE.Match(text) {
+				return errors.New(fmt.Sprintf("Rule %s: failed to match value (%s, %s)", r.name, r.matchRE.String(), string(text)))
+			}
+		}
+	}
+	if resultCount == 0 && !r.acceptNoMatches {
+		return errors.New(fmt.Sprintf("Rule %s failed: No matches", r.name))
+	}
+
+	return nil
+}
+
+type jsonPath struct {
```

Comment:
have this embed *admission.Handler, see example here:

https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/admission/resourcequota/admission.go#L68

Then you don't need to implement `Handles` (or anything else that may come in the future)